tools/acrn-crashlog/acrnprobe/conf.rst

```diff
@@ -0,0 +1,379 @@
+.. _acrnprobe-conf:
+
+Configuration Of Acrnprobe
+##########################
+
+Description
+***********
+``Acrnprobe`` uses XML as the format of its configuration file, namely
+``acrnprobe.xml``, so the configuration file should reach the `XML standard`_
+at least.
+
+Layout
+******
+
+.. code-block:: xml
+
+   <?xml version="1.0" encoding="UTF-8"?>
+   <conf>
+   Root node of configuration.
+
+   <senders>
+   Configuration section of senders.
+   </senders>
+
+   <triggers>
+   Configuration section of triggers.
+   </triggers>
+
+   <vms>
+   Configuration section of virtual machines.
+   </vms>
+
+   <logs>
+   Configuration section of logs.
+   </logs>
+
+   <crashes>
+   Configuration section of crashes.
+   Note that this section should be configured after triggers and logs, as
+   crashes depend on these two sections.
+   </crashes>
+
+   <infos>
+   Configuration section of infos.
+   Note that this section should be configured after triggers and logs, as
+   infos depend on these two sections.
+   </infos>
+
+   </conf>
+
+As for the definition of ``sender``, ``trigger``, ``crash`` and ``info`` and
+information of supported ``vm``, please refer to :ref:`acrnprobe_doc`.
+
+Properties of modules
+*********************
+
+Common properties
+=================
+
+- ``id``:
+  The index, which grows from 1, in its group.
+- ``enable``:
+  This part of configuration can only take effect when the value is ``true``.
+
+Other properties
+================
+
+- ``inherit``:
+  Specify a parent for a certain crash.
+  The child crash will inherit all configurations from the specified (by id)
+  crash. These inherited configurations could be overwritten by new ones.
+  Also, this property helps build the crash tree in ``acrnprobe``.
+- ``expression``:
+  See `Crash`_.
+
+Crash tree in acrnprobe
+***********************
+
+There could be a parent-child relationship between crashes. Refer to the
+diagrams below, crash B and D are the children of crash A, because crash B and
+D inherit from crash A, and crash C is the child of crash B.
+
+Build crash tree in configuration
+=================================
+
+.. graphviz::
+
+   digraph {
+      {
+         node [shape=plaintext];
+         "level 1" -> "level 2" -> "level 3";
+      }
+
+      node [shape=box;style="rounded,filled";color=AntiqueWhite;];
+      c1 [ label="crash A\nid 1\ncrash root" ];
+      c2 [ label="crash B\nid 2" ];
+      c3 [ label="crash C\nid 3\ncrash leaf" ];
+      c4 [ label="crash D\nid 4\ncrash leaf" ];
+      c5 [ label="crash E\nid 5\ncrash root\ncrash leaf" ];
+      { rank = same; "level 1"; c1; c5;}
+      { rank = same; "level 2"; c2; c4;}
+      { rank = same; "level 3"; c3;}
+
+      node [shape=box;color="transparent";];
+      "None" -> {c1 c5} [ label="inherit 0" ];
+      c1 -> {c2 c4} [ label="inherit 1" ];
+      c2 -> c3 [ label="inherit 2" ];
+   }
+
+Crash type matches at runtime
```

Comment:
What is a "crash type"?  Is there a list of crash types defined somewhere?

Reply:
Each crash has its own ``crash type``, which equals the ``name`` of crash.
How about  changing ``Crash type matches at runtime`` to ``Match crash at runtime``